app/models/article.rb

```diff
@@ -78,6 +78,8 @@ class Article < ApplicationRecord
 
   scope :published, -> { where(published: true) }
   scope :unpublished, -> { where(published: false) }
+  scope :active, -> { where(archived: false) }
+  scope :active_published_articles, -> { published.active }
```

Comment:
I'm not convinced about the naming, I wouldn't add the concept of `active`, it's probably not correct to consider published articles that are not archived as "active". 

My suggestion is to add only one scope, `not_archived`, and then in the controller use `Article.published.not_archived` which is clearer.

Rails 6 AFAIK adds negative scopes but we're not there yet, so we have to add it manually :)